shieldapi/__init__.py

```diff
@@ -0,0 +1,29 @@
+import logging
+import os
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger("shieldapi")
+
+
+def check_mandatory_env_vars():
+    """
+    Checks if all mandatory environment variables are set.
+    Raises an EnvironmentError if any are missing.
+    """
+    mandatory_vars = [
+        "KEYCLOAK_HOST",
+        "KEYCLOAK_REALM_NAME",
+        "KEYCLOAK_CLIENT_ID",
+        "KEYCLOAK_CLIENT_SECRET",
+    ]
+
+    missing_vars = [var for var in mandatory_vars if not os.environ.get(var)]
+
+    if missing_vars:
+        logger.critical(
+            "Missing mandatory environment variables: %s", ", ".join(missing_vars)
+        )
+
+
+check_mandatory_env_vars()
```

Comment:
@yoavnash , would it makes sense to call it "optional" envs instead of mandatory?  Since they can also be set via kwargs, I would not call them mandatory.

Reply:
As separately discussed, the idea here is to have an automatic check whether the required env vars are set. The check can possibly be extended to cover also the case of kwargs.

Reply:
We are already providing this warning [here](https://github.com/materials-marketplace/shieldapi/blob/0710a573ff5569c056bbdfeb1f91ce8ca29b5b79/shieldapi/keycloak_utils.py#L125) in case of the kwarg is set. Hence, I would not make a `logger.critial` but rather a `warnings.debug` and change the text of the message slightly since it the envs are not necessarily critical when the kwarg is provided.

Reply:
If you check kwargs additionally, then I think `logger.critial` makes sense because the library can't function without them.
We should also ensure that these checks are done in the startup of the service and not in the first call to keycloak.

Reply:
Decision: we make it as `logger.warning`. Possibly add a way to skip this check in order to mute the warnings.